src/pages/instances/forms/ConvertToInstanceForm.tsx

```diff
@@ -0,0 +1,179 @@
+import { ChangeEvent, FC, useCallback } from "react";
+import {
+  ActionButton,
+  Button,
+  Icon,
+  Input,
+  Modal,
+  Select,
+} from "@canonical/react-components";
+import { Form } from "react-router-dom";
+import { FormikProps } from "formik";
+import { UploadState } from "types/storage";
+import ProgressBar from "components/ProgressBar";
+import { getFileExtension, humanFileSize } from "util/helpers";
+import StoragePoolSelector from "pages/storage/StoragePoolSelector";
+import {
+  ConvertToInstanceFormValues,
+  supportedVMArchOptions,
+} from "util/convertToInstance";
+import { sanitizeInstanceName, truncateInstanceName } from "util/instances";
+
+interface Props {
+  close: () => void;
+  formik: FormikProps<ConvertToInstanceFormValues>;
+  isLoading: boolean;
+  disableSubmit: boolean;
+  project: string;
+  uploadState: UploadState | null;
+}
+
+const ConvertToInstanceForm: FC<Props> = ({
+  close,
+  formik,
+  isLoading,
+  disableSubmit,
+  project,
+  uploadState,
+}) => {
+  const handleFileChange = async (e: ChangeEvent<HTMLInputElement>) => {
+    const { onChange } = formik.getFieldProps("imageFile");
+    onChange(e);
+
+    if (e.currentTarget.files) {
+      const file = e.currentTarget.files[0];
+      const suffix = "-instance";
+      const fileExtension = getFileExtension(file.name);
```

Comment:
This function doesn't really return the file extension, but an array. Maybe better to bundle this logic together with  sanitize instance name. Instead of splitting the work over two functions.

Reply:
`getFileExtension` returns a string? Not sure about lumping the logic with `sanitizeInstanceName` since not all instance names might need a split by file extension

Reply:
What I meant is: from a function called `getFileExtension(input: string)` I'd expect to get a string back, not an array.

Reply:
but `getFileExtension` returns a string, see below code for the function.

```
export const getFileExtension = (filename: string): string => {
  if (!filename.includes(".")) {
    return "";
  }

  return `.${filename.split(".").pop()}` || "";
};
```

Reply:
Aah I see, no problem :+1: